### PR TITLE
DRAFT : New component : "Autres préciser" / "Other" (Checkbox, checkBoxOne)

### LINKED ISSUE #204: New component : "Autres préciser" / "Other" (Checkbox, checkBoxOne)   (11/2022)
New component : creation of a textual complement to specify the choice of a modality of type "Other" (in checkBox).
 

=> https://github.com/InseeFr/Bowie/issues/5

### PR DESCRIPTION
**Pull request by Grafikart**
## Issue

On some form we want to add an option to allow an additional answer for specific field

## Solution

There is not lunatic modelisation for this feature so this PR is a proof of concept and spec. The field is triggered adding extra properties on the "responses" field : 

```diff
{
	"id": "kmort6x9-QOP-kmorue9d",
	"label": {
		"value": "\"Autre\"",
		"type": "VTL"
	},
	"response": { "name": "NATIO1N5" },
+	"detail": {
+		"label": {
+			"value": "\"Préciser : \"",
+			"type": "VTL"
+		},
+		"response": {
+			"name": "NATIO1N5DETAIL"
+		}
+	}
}
```

## Render 

image

## Extra

While working on this feature I refactored the CheckboxGroup component to use a native checkbox input as a base.

## Breaking changes

I did a big refactor while working on this MR to group everything inside the "CheckboxGroup" component. This component is used in both CheckboxOne, CheckboxGroup, Radio (since the behaviour is pretty similar between this component) so it could require some tweaking.

Fix #204

### CODE REVIEW

File: src/components/checkbox/commons/checkbox-option.tsx [hunk 1]
```diff
@@ -1,95 +1,103 @@
...
-					<Label id={labelId} htmlFor={id} description={description}>
-						{codeModality && (
-							<span className="code-modality">
-								{codeModality.toUpperCase()}
-							</span>
-						)}
-						{label}
-					</Label>
-				</span>
+					onChange={(e) => handleChange(e.target.checked)}
```
Comment (laurentC35) [inline src/components/checkbox/commons/checkbox-option.tsx:100]: La fonction est définie ici, un re-rendu inutile peut-être ?

**Review by laurentC35 — CHANGES_REQUESTED** [review]
Dans la story sur le autre précisez, il manque la variable `NATIO1N5DETAIL` dans le tableau des variables COLLECTED.

[Some unit test, don't succeed](https://github.com/InseeFr/Lunatic/actions/runs/6732228822/job/18298579048?pr=750)

**Comment by nsenave** [comment]
I'm ok with this modelling, I would just need to know the comprehensive list of components that can have this `"detail"` property. 

Is this only for `CheckboxGroup`? or also for `CheckboxBoolean` and `Radio`. And if so:

- what about `Dropdown`?
- what about the case when these components would be in a `Table` (would it be forbidden, or...?)

PS: on the naming, the idea of `"otherPleaseSpecify"` came to my mind (longer, a bit weird, but a bit clearer imo, just an idea though)

**Comment by nsenave** [comment]
Discussed with @AnneHuSKa : 

- the "detail" is not expected to work if the component is within a table.
- we also want the "other, please specify" feature in `Radio` and `CheckboxOne` components
- we don't want the "other, please specify" in the `Dropdown` case

File: src/utils/event.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type { KeyboardEvent, KeyboardEventHandler } from 'react';
+
+export function keyHandler<T>(key: string, handler: KeyboardEventHandler<T>) {
+	return (e: KeyboardEvent<T>) => {
+		console.log(e.key);
```
Comment (laurentC35) [inline src/utils/event.ts:5]: You should remove console.log ?

**Comment by ddecrulle** [comment]
Completed in another PR

**Pull request closed without merging**